CI: 3.8 numpy_dev build failing recently

https://dev.azure.com/pandas-dev/pandas/_build/results?buildId=54353&view=logs&j=eab14f69-13b6-5db7-daeb-7b778629410b&t=ce687173-08c6-5301-838d-71b2dda24510

```
2021-02-16T22:44:34.3078267Z =================================== FAILURES ===================================
2021-02-16T22:44:34.3078772Z ___________________ TestTableOrientReader.test_comprehensive ___________________
2021-02-16T22:44:34.3079881Z [gw1] linux -- Python 3.8.5 /home/vsts/miniconda3/envs/pandas-dev/bin/python
2021-02-16T22:44:34.3080157Z 
2021-02-16T22:44:34.3080536Z self = <pandas.tests.io.json.test_json_table_schema.TestTableOrientReader object at 0x7f8908515790>
2021-02-16T22:44:34.3080847Z 
2021-02-16T22:44:34.3081136Z     def test_comprehensive(self):
2021-02-16T22:44:34.3081478Z         df = DataFrame(
2021-02-16T22:44:34.3081758Z             {
2021-02-16T22:44:34.3082075Z                 "A": [1, 2, 3, 4],
2021-02-16T22:44:34.3082433Z                 "B": ["a", "b", "c", "c"],
2021-02-16T22:44:34.3083091Z                 "C": pd.date_range("2016-01-01", freq="d", periods=4),
2021-02-16T22:44:34.3083753Z                 # 'D': pd.timedelta_range('1H', periods=4, freq='T'),
2021-02-16T22:44:34.3084228Z                 "E": pd.Series(pd.Categorical(["a", "b", "c", "c"])),
2021-02-16T22:44:34.3084712Z                 "F": pd.Series(pd.Categorical(["a", "b", "c", "c"], ordered=True)),
2021-02-16T22:44:34.3085138Z                 "G": [1.1, 2.2, 3.3, 4.4],
2021-02-16T22:44:34.3085802Z                 "H": pd.date_range("2016-01-01", freq="d", periods=4, tz="US/Central"),
2021-02-16T22:44:34.3086267Z                 "I": [True, False, False, True],
2021-02-16T22:44:34.3086608Z             },
2021-02-16T22:44:34.3086937Z             index=pd.Index(range(4), name="idx"),
2021-02-16T22:44:34.3087246Z         )
2021-02-16T22:44:34.3087499Z     
2021-02-16T22:44:34.3087786Z >       out = df.to_json(orient="table")
2021-02-16T22:44:34.3087971Z 
2021-02-16T22:44:34.3088300Z pandas/tests/io/json/test_json_table_schema.py:772: 
2021-02-16T22:44:34.3088746Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2021-02-16T22:44:34.3089152Z pandas/core/generic.py:2499: in to_json
2021-02-16T22:44:34.3089486Z     return json.to_json(
2021-02-16T22:44:34.3090018Z pandas/io/json/_json.py:115: in to_json
2021-02-16T22:44:34.3090343Z     s = writer(
2021-02-16T22:44:34.3090703Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2021-02-16T22:44:34.3090961Z 
2021-02-16T22:44:34.3091280Z self = <pandas.io.json._json.JSONTableWriter object at 0x7f8908515f40>
2021-02-16T22:44:34.3091597Z 
2021-02-16T22:44:34.3091857Z     def write(self):
2021-02-16T22:44:34.3092200Z         iso_dates = self.date_format == "iso"
2021-02-16T22:44:34.3092519Z >       return dumps(
2021-02-16T22:44:34.3092827Z             self.obj_to_write,
2021-02-16T22:44:34.3093139Z             orient=self.orient,
2021-02-16T22:44:34.3093492Z             double_precision=self.double_precision,
2021-02-16T22:44:34.3093853Z             ensure_ascii=self.ensure_ascii,
2021-02-16T22:44:34.3094287Z             date_unit=self.date_unit,
2021-02-16T22:44:34.3094747Z             iso_dates=iso_dates,
2021-02-16T22:44:34.3095093Z             default_handler=self.default_handler,
2021-02-16T22:44:34.3095449Z             indent=self.indent,
2021-02-16T22:44:34.3095724Z         )
2021-02-16T22:44:34.3096244Z E       DeprecationWarning: an integer is required (got type float).  Implicit conversion to integers using __int__ is deprecated, and may be removed in a future version of Python.
2021-02-16T22:44:34.3096661Z 
2021-02-16T22:44:34.3096974Z pandas/io/json/_json.py:177: DeprecationWarning

```

--- a/pandas/tests/io/json/test_json_table_schema.py
+++ b/pandas/tests/io/json/test_json_table_schema.py
@@ -753,6 +753,9 @@ class TestTableOrientReader:
         result = pd.read_json(out, orient="table")
         tm.assert_frame_equal(df, result)
 
+    @pytest.mark.filterwarnings(
+        "ignore:an integer is required (got type float)*:DeprecationWarning"
+    )
     def test_comprehensive(self):
         df = DataFrame(
             {

